pkg/pool-utils/contracts/BasePoolFactory.sol

```diff
@@ -117,7 +134,20 @@ abstract contract BasePoolFactory is IBasePoolFactory, SingletonAuthentication,
     }
 
     function _create(bytes memory constructorArgs, bytes32 salt) internal returns (address pool) {
-        pool = CREATE3.deploy(_computeFinalSalt(salt), abi.encodePacked(_creationCode, constructorArgs), 0);
+        bytes memory creationCode = abi.encodePacked(_creationCode, constructorArgs);
+        bytes32 finalSalt = _computeFinalSalt(salt);
+        assembly {
+            pool := create2(0, add(creationCode, 32), mload(creationCode), finalSalt)
```

Comment:
Same here: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/Create2.sol. This low level bit is abstracted by `deploy`.

I understand you still need to pack the constructor args with the library.

Reply:
Done!